bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClientFactory.java

```diff
@@ -145,9 +147,16 @@ public boolean equals(Object o) {
   }
 
   private BigQueryReadClient createBigQueryReadClient(
-      Optional<String> endpoint, Optional<Integer> flowControlWindow) {
+      Optional<String> endpoint,
+      Optional<Integer> channelPoolSize,
+      Optional<Integer> flowControlWindow) {
     try {
       InstantiatingGrpcChannelProvider.Builder transportBuilder = createTransportBuilder(endpoint);
+      if (channelPoolSize.isPresent()) {
```

Comment:
as I think we likely want to be more sophisticated on the default here, and use number of cores from spark (and possibly multiply be streams per partition)